server/src/internalClusterTest/java/org/elasticsearch/action/admin/indices/create/CreateSystemIndicesIT.java

```diff
@@ -112,6 +198,38 @@ private void doCreateTest(Runnable runnable, String concreteIndex) {
         ensureGreen(INDEX_NAME);
 
         assertMappingsAndSettings(TestSystemIndexDescriptor.getNewMappings(), concreteIndex);
+        assertAliases(concreteIndex);
+    }
+
+    /**
+     * Make sure that aliases are created hidden
+     */
+    private void assertAliases(String concreteIndex) {
+        final GetAliasesResponse getAliasesResponse = client().admin()
+            .indices()
+            .getAliases(new GetAliasesRequest().indicesOptions(IndicesOptions.strictExpandHidden()))
+            .actionGet();
+
+        assertThat(getAliasesResponse.getAliases().size(), equalTo(1));
+        assertThat(getAliasesResponse.getAliases().get(concreteIndex).size(), equalTo(1));
+        assertThat(getAliasesResponse.getAliases().get(concreteIndex).get(0).isHidden(), equalTo(true));
+    }
+
+    private void assertAliasesAddedByTemplate(Set<String> aliasNames) throws InterruptedException, java.util.concurrent.ExecutionException {
```

Comment:
Nit: This method is confusingly named - I spent a while trying to figure out why `".test-index"` was in the set when this gets called. Maybe `assertHasAliases`?